### ISSUE TITLE
feat: Tabular model package — high-level Train/Predict/Save/Load API for structured data

### ISSUE THREAD

**Issue by dndungu** [zerfoo/zerfoo#102]
## Summary

Training tabular (structured) data models currently requires manually constructing compute graphs, writing training loops, computing losses, and handling serialization. A high-level \`tabular\` package would make this accessible and production-ready.

This is the most impactful feature for Wolf (Zerfoo's primary tabular consumer) — it would replace ~300 lines of hand-rolled training code with a clean API.

## Proposed API

```go
package tabular

// Model represents a trained tabular model.
type Model[T numeric.Float] struct { ... }

// Config defines the model architecture and training hyperparameters.
type Config struct {
    Architecture string  // \"mlp\", \"ft-transformer\", \"tabnet\", \"resnet\"
    HiddenDim    int     // auto-sized if 0
    NumLayers    int     // default 2
    Dropout      float64 // default 0.3
    LR           float64 // default 0.001
    Epochs       int     // default 100
    BatchSize    int     // default 64
    EarlyStopping int   // patience (0 = disabled)
    Task         string  // \"binary_classification\" or \"regression\"
}

// Train creates and trains a tabular model.
func Train[T numeric.Float](
    engine compute.Engine[T],
    data [][]float64,  // features + label (last col)
    config Config,
) (*Model[T], TrainResult, error)

// Predict runs inference on new data.
func (m *Model[T]) Predict(features [][]float64) ([]float64, error)

// Save/Load for model persistence.
func (m *Model[T]) Save(dir string) error
func Load[T numeric.Float](dir string, engine compute.Engine[T]) (*Model[T], error)

// Ensemble combines multiple models.
func Ensemble[T numeric.Float](models []*Model[T], weights []float64) *Model[T]
```

## Use Case

Wolf's complete training pipeline would reduce to:

```go
engine := compute.NewCPUEngine[float32](numeric.Float32Ops{})
model, result, _ := tabular.Train(engine, data, tabular.Config{
    Architecture: \"mlp\",
    Task:         \"binary_classification\",
    Epochs:       100,
    Dropout:      0.3,
    EarlyStopping: 10,
})
preds, _ := model.Predict(testData)
model.Save(\"models/COIN-15m/v1\")
```

## Components Required

This depends on several other issues:
- #94: LoadParameters (for Load)
- #95: BCE loss (for binary classification)
- #97: Dropout
- #98: Sigmoid
- #99: BatchNorm
- #100: Parameter ordering guarantee
- #96: Save/Load

This is the capstone feature that ties them all together.
